packages/typespec-client-generator-core/src/license.ts

```diff
@@ -0,0 +1,139 @@
+\n
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n`,
+  },
+  "MPL 2.0": {
+    name: "MPL 2.0",
+    link: "https://www.mozilla.org/en-US/MPL/2.0/",
+    company: "",
+    description: `Copyright © <company>\n
+\n
+This Source Code Form is subject to the terms of the Mozilla Public\n
+License, v. 2.0. If a copy of the MPL was not distributed with this\n
+file, You can obtain one at https://mozilla.org/MPL/2.0/.\n`,
+  },
+  "GPL-3.0": {
+    name: "GPL-3.0",
+    link: "https://www.gnu.org/licenses/gpl-3.0.html",
+    company: "",
+    description: `Copyright © <company>\n
+\n
+This program is free software: you can redistribute it and/or modify\n
+it under the terms of the GNU General Public License as published by\n
+the Free Software Foundation, either version 3 of the License, or\n
+(at your option) any later version.\n
+\n
+This program is distributed in the hope that it will be useful,\n
+but WITHOUT ANY WARRANTY; without even the implied warranty of\n
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the\n
+GNU General Public License for more details.\n
+\n
+You should have received a copy of the GNU General Public License\n
+along with this program.  If not, see <https://www.gnu.org/licenses/>.\n`,
+  },
+  "LGPL-3.0": {
+    name: "LGPL-3.0",
+    link: "https://www.gnu.org/licenses/lgpl-3.0.html",
+    company: "",
+    description: `Copyright © <company>\n
+\n
+This program is free software: you can redistribute it and/or modify\n
+it under the terms of the GNU Lesser General Public License as published by\n
+the Free Software Foundation, either version 3 of the License, or\n
+(at your option) any later version.\n
+\n
+This program is distributed in the hope that it will be useful,\n
+but WITHOUT ANY WARRANTY; without even the implied warranty of\n
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the\n
+GNU Lesser General Public License for more details.\n
+\n
+You should have received a copy of the GNU Lesser General Public License\n
+along with this program.  If not, see <https://www.gnu.org/licenses/>.\n`,
+  },
+};
+
+export function getLicenseInfo(context: TCGCContext): LicenseInfo | undefined {
+  if (!context.licenseName || context.licenseName.trim() === "") {
+    return undefined;
+  }
```

Comment:
It seems licenseName is necessary however users may not know it. So before return undefined, maybe we could report a warning diagnostics like "licenseName is necessary"

Reply:
if there is no `licenseName` configured, we will leave the license part as empty.